spring-integration-core/src/main/java/org/springframework/integration/config/MessagingGatewayRegistrar.java

```diff
@@ -72,6 +82,13 @@ public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata, B
 	}
 
 	public BeanDefinitionHolder parse(Map<String, Object> gatewayAttributes) {
+		boolean completableFutureCapable = true;
+		try {
+			ClassUtils.forName("java.util.concurrent.CompletableFuture", this.beanClassLoader);
```

Comment:
Since this one is a part of JVM I don't see reason to worry about classloader.
Although you have already done that we can live with that and merge as is.